pkg/build/util/util.go

```diff
@@ -238,17 +239,22 @@ func SafeForLoggingURL(u *url.URL) *url.URL {
 	if u == nil {
 		return nil
 	}
-	newUrl := *u
-	newUrl.User = url.User("redacted")
-	return &newUrl
+	newURL, err := url.Parse(u.String())
```

Comment:
url doesn't support `DeepCopy()`, so I re-parsed the string instead.

Reply:
that seems fine